src/main/java/com/mentoring/ecommerce/adapter/in/web/mapper/ProductMapper.java

```diff
@@ -4,8 +4,10 @@
 import com.mentoring.ecommerce.adapter.in.web.response.ProductResponse;
 import com.mentoring.ecommerce.domain.Product;
 import org.mapstruct.Mapper;
+import org.springframework.stereotype.Service;
 
 @Mapper(componentModel = "spring")
+@Service
```

Comment:
To make the mapping work, I needed to declare it as `@Component`, but just doing that wasn't enough, so I tried `@Service` and it worked fine. Even though these annotations are quite similar, only @Service worked.

**Problem:** Mapping was not working, returning null for both `request` and `response` body, even though the query was working fine.

Reply:
That's interesting! I'll search latter why `@Component` did not work.